.github/workflows/e2e.yml

```diff
@@ -0,0 +1,87 @@
+name: e2e
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 4 * * *'
+  pull_request:
+    branches:
+      - main
+      - 'release-*'
+  push:
+    branches:
+      - main
+      - 'release-*'
+
+permissions:
+  contents: read
+
+jobs:
+  e2e:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: e2e-dsa-spr
+            targetjob: e2e-dsa
+            runner: simics-spr
+            images: intel-dsa-plugin intel-idxd-config-initcontainer accel-config-demo
+          - name: e2e-dsa-gnr
+            targetjob: e2e-dsa
+            runner: simics-gnr
+            images: intel-dsa-plugin intel-idxd-config-initcontainer accel-config-demo
+          - name: e2e-dlb-spr
+            targetjob: e2e-dlb
+            runner: simics-spr
+            images: intel-dlb-plugin intel-dlb-initcontainer dlb-libdlb-demo
+          - name: e2e-dlb-gnr
+            targetjob: e2e-dlb
+            runner: simics-gnr
+            images: intel-dlb-plugin intel-dlb-initcontainer dlb-libdlb-demo
+          - name: e2e-fpga
+            runner: fpga
+            images: intel-fpga-plugin intel-fpga-initcontainer intel-fpga-admissionwebhook opae-nlb-demo
+          - name: e2e-gpu
+            runner: gpu
+            images: intel-gpu-plugin intel-gpu-initcontainer
+          - name: e2e-iaa-spr
+            targetjob: e2e-iaa
+            runner: simics-spr
+            images: intel-iaa-plugin intel-idxd-config-initcontainer accel-config-demo
+          - name: e2e-iaa-gnr
+            targetjob: e2e-iaa
+            runner: simics-gnr
+            images: intel-iaa-plugin intel-idxd-config-initcontainer accel-config-demo
+          - name: e2e-qat
+            runner: qat
+            images: intel-qat-plugin intel-qat-initcontainer crypto-perf
+          - name: e2e-qat4
+            runner: simics-spr
+            images: intel-qat-plugin intel-qat-initcontainer openssl-qat-engine
+          - name: e2e-sgx
+            runner: sgx
+            images: intel-sgx-plugin intel-sgx-initcontainer intel-sgx-admissionwebhook
+
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.runner }}
+    env:
+      TARGET_JOB: ${{ matrix.targetjob || matrix.name }}
```

Comment:
could we speed up things if we move to a model where several targets are run on the same runner:
```
    strategy:
      fail-fast: true
      matrix:
        runner: [fpga, gpu, qat, sgx, simics-gnr, simics-spr]
        include:
          - runner: fpga
            target: e2e-fpga
          - runner: gpu
            target: e2e-gpu
          - runner: qat
            target: e2e-qat
          - runner: sgx
            target: e2e-sgx
          - runner: simics-gnr
            target: e2e-dlb e2e-dsa e2e-iaa
          - runner: simics-spr
            target: e2e-dlb e2e-dsa e2e-iaa e2e-qat4
```

Reply:
do you mean one after another, without re-provisioning?

Reply:
yes: `make e2e-dlb e2e-dsa e2e-iaa`

Reply:
This would be against our CI design - to test changes in the clean environment.
However, having an additional test that runs multiple workloads sequentially could be a good idea.